aiida_kkr/cmdline/util/options.py

```diff
@@ -0,0 +1,128 @@
+# -*- coding: utf-8 -*-
+"""
+Options commonly used throughout the aiida-kkr command line interface.
+To standardize at least of the options, they are kept as close as possible to
+aiida-fleur. aiida-core and aiida-quantumespresso
+"""
+import click
+from aiida.cmdline.params import types
+from aiida.cmdline.params.options import OverridableOption
+from .types import StructureNodeOrFileParamType
+from .defaults import get_voro, get_kkr, get_kkrimp
+
+STRUCTURE_OR_FILE = OverridableOption(
+    '-s',
+    '--structure',
+    type=StructureNodeOrFileParamType(),
+    help='StructureData node, given by pk or uuid or file in any for mat which will be converted.')
+
+STRUCTURE = OverridableOption('-s',
+                              '--structure',
+                              type=types.DataParamType(sub_classes=('aiida.data:structure',)),
+                              help='StructureData node, given by pk or uuid.')
+
+VORO = OverridableOption('-v',
+                           '--voro',
+                           type=types.CodeParamType(entry_point='kkr.voro'),
+                           default=get_voro,
+                           show_default=True,
+                           help='A code node or label for an voronoi executable.')
+
+KKR = OverridableOption('-k',
+                          '--kkr',
+                          type=types.CodeParamType(entry_point='kkr.kkr'),
+                          default=get_kkr,
+                          show_default=True,
+                          help='A code node or label for a KKR executable.')
+
+KKRIMP = OverridableOption('-ki',
+                          '--kkr-imp',
+                          type=types.CodeParamType(entry_point='kkr.kkrimp'),
+                          default=get_kkrimp,
+                          show_default=True,
+                          help='A code node or label for a KKRimp executable.')
+
+PARAMETERS = OverridableOption(
+    '-p',
+    '--parameters',
+    type=types.DataParamType(sub_classes=('aiida.data:dict',)),
+    help='Dict with calculation parameters which will be given to the calculation or workchain.')
+
+KPOINTS = OverridableOption(
+    '-kpt',
+    '--kpoints',
+    type=types.DataParamType(sub_classes=('aiida.data:array.kpoints',)),
+    help='An aiida KpointsData node.')
+
+SETTINGS = OverridableOption('-set',
```

Comment:
What is a settings node for the calcjob? The calcjobs of AiiDA-KKR use calculation parameters (Dict) node which are the `PARAMETERS` option and workflows use the workflow parameters (`WF_PARAMETERS` option).

Reply:
aiida-fleur, and qe use settings to allow the user to provide additional execute options or add files retrieved or not retrieved, etc. I never used this for my calcs in fleur, but if one wants the full functionality is it needed. Of course if the kkr plugin does not have this, it should be deleted.